man/fabric.7.md

```diff
@@ -238,6 +238,65 @@ FI_LOG_SUBSYS environment variables.
 - *mr*
 : Provides output specific to memory registration.
 
+# FILE SYSTEM CONFIGRATION INTERFACE
+
+Libfabric defines a set of file system interfaces that may be used to
+set global or provider specific parameters.  Conceptually, this interface
+is similar to the /proc interfaces used by ip(7) and tcp(7).
+Parameters can be accessed by reading or writing files in
+%SYSCONFDIR%/rdma/fabric.  The files are typically found at
+/etc/rdma/fabric or /usr/local/etc/rdma/fabric.
+The exact location of the files is controllable through build parameters,
+but for simplicity, the following documentation assumes that they are found
+under /etc/rdma/fabric.  The files do not exist by default and must be
+created by the administrator or Linux distro.  
+
+Parameter files located under /etc/rdma/fabric are treated as global and
+apply to all relevant providers.  Provider specific parameters may be set
+by writing to a file by the same name, but in a provider specific
+subdirectory.  For example, parameter files under /etc/rdma/fabric/verbs
+will apply only to the verbs provider.
+
+Note that providers may select alternate values that those
+specified.  This may be necessary in order to adhere to memory alignment
+restrictions, hardware maximums, or protocol limitations, for example.
+
+Most files are associated with a single value, with the data type
+shown in the descriptions below.  Administrators can set and get the values
+using a method similar to this example:
+
+% mkdir /etc/rdma
+% mkdir /etc/rdma/fabric
+% echo 64 > /etc/rdma/fabric/def_inject_size
+% cat /etc/rdma/fabric/def_inject_size
+  64
+
+In some cases, system files may conflict with libfabric or provider
+environment variables.  In most cases, if both values are provided, the
+environment variable takes precedence.  The exception are that parameter
+files that specify a maximum will limit environment variable settings.
```

Comment:
> I think param files are fine, and Env vars are fine. But the usage model
> needs to be clear and unambiguous. E.G., open MPI supports multiple
> methods of setting config vars. But *all* config vars are settable through
> *all* methods (e.g., file, Env var, etc). That way, everything is equal.

I am going to tie this into the existing param framework that we already have (src/var.c).  This will allow the user to set all values either through param files or env vars.  Plus, the documentation becomes automatic.

The precedence is now clearly defined with the param files having higher precedence.

Thanks for the feedback.